app/src/main/java/com/tpp/theperiodpurse/ui/cycle/CycleScreen.kt

```diff
@@ -100,14 +117,25 @@ fun CycleHistoryBox(modifier: Modifier = Modifier) {
                     .padding(vertical = 15.dp)
                     .fillMaxWidth()
             )
-            Text(text = stringResource(R.string.log_to_learn))
+            Text(text = "TODO")
         }
     }
 }
 
 @Composable
-fun CycleScreenLayout(modifier: Modifier = Modifier) {
+fun CycleScreenLayout(
+    appViewModel: AppViewModel,
+    calendarViewModel: CalendarViewModel,
+    modifier: Modifier = Modifier
+) {
+    val dates = ArrayList(appViewModel.getDates())
+
+    periodLength = calculateAveragePeriodLength(dates)
+    cycleLength = calculateAverageCycleLength(dates)
+
     ThePeriodPurseTheme {
+        appViewModel.loadData(calendarViewModel)
```

Comment:
I'm not too sure how compose handles recomposing, but is it necessary to load the data every time, or does `calendarViewModel` already has the most up to date data?

Reply:
I'm not entirely sure how all of it works, but I think with how the `AppViewModel.loadData` function behaves, I need it to update every time there's a change. I can make it as a `LaunchedEffect` so that it knows to only load the data when composition starts, though.

Reply:
I previously was having issues with the dates not being updated which is why I included the `calendarViewModel` which fixed my problem at the time, but I think you're right. I can make these changes :)